homeassistant/components/mopeka/__init__.py

```diff
@@ -37,9 +42,15 @@ async def async_setup_entry(hass: HomeAssistant, entry: MopekaConfigEntry) -> bo
     await hass.config_entries.async_forward_entry_setups(entry, PLATFORMS)
     # only start after all platforms have had a chance to subscribe
     entry.async_on_unload(coordinator.async_start())
+    entry.async_on_unload(entry.add_update_listener(update_listener))
     return True
 
 
-async def async_unload_entry(hass: HomeAssistant, entry: MopekaConfigEntry) -> bool:
+async def update_listener(hass: HomeAssistant, entry: MopekaConfigEntry) -> None:
+    """Handle options update."""
+    await hass.config_entries.async_reload(entry.entry_id)
+
+
+async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
```

Comment:
```suggestion
async def async_unload_entry(hass: HomeAssistant, entry: MopekaConfigEntry) -> bool:
```

We want to use the integration specific type for all instances where the config entry is a mokepa config entry

Reply:
https://developers.home-assistant.io/blog/2024/04/30/store-runtime-data-inside-config-entry